src/views/Root/GroceryService.js

```diff
@@ -0,0 +1,67 @@
+import { localStorageKeys } from './StorageKeys';
+
+export class GroceryService {
+	fetchArrayByKey(key) {
+		const groceries = localStorage.getItem(key);
+		return groceries ? JSON.parse(groceries) : [];
+	};
+
+	setByKey(key, value) {
+		localStorage.setItem(key, JSON.stringify(value));
+	};
+
+	getAllGroceries() {
+		return this.fetchArrayByKey(localStorageKeys.grocery);
+	};
+
+	getAllCompletedGroceries() {
+		return this.fetchArrayByKey(localStorageKeys.groceryCompleted);
+	};
+
+	addGroceryItem(newItem) {
+		const grocery = this.fetchArrayByKey(localStorageKeys.grocery);
+
+		grocery.push(newItem);
```

Comment:
Work with immutable data and operations. Leads to easier to reason about and easier to maintain code.